tests/Doctrine/Tests/DBAL/Types/BinaryTest.php

```diff
@@ -52,19 +52,34 @@ public function testBinaryNullConvertsToPHPValue()
 
     public function testBinaryStringConvertsToPHPValue()
     {
-        $databaseValue = 'binary string';
+        $databaseValue = $this->getBinaryString();
         $phpValue      = $this->type->convertToPHPValue($databaseValue, $this->platform);
 
-        self::assertInternalType('resource', $phpValue);
-        self::assertEquals($databaseValue, stream_get_contents($phpValue));
+        self::assertSame($databaseValue, $phpValue);
     }
 
     public function testBinaryResourceConvertsToPHPValue()
     {
         $databaseValue = fopen('data://text/plain;base64,' . base64_encode('binary string'), 'r');
         $phpValue      = $this->type->convertToPHPValue($databaseValue, $this->platform);
 
-        self::assertSame($databaseValue, $phpValue);
+        self::assertSame('binary string', $phpValue);
+    }
+
+    /**
+     * Creates a binary string containing all possible byte values.
+     *
+     * @return string
+     */
+    private function getBinaryString()
+    {
+        $string = '';
+
+        for ($i = 0; $i < 256; $i++) {
```

Comment:
`return implode(array_map('chr', range(1, 256)))`? Or am I pushing it too far? :D